code/modules/projectiles/guns/energy/plasma.dm

```diff
@@ -40,12 +40,12 @@
 	fire_delay = 20
 
 	firemodes = list(
-		list(mode_name="DESTROY", projectile_type=/obj/item/projectile/beam/pulse, fire_sound='sound/weapons/pulse.ogg', fire_delay=null, charge_cost=null, icon="destroy"),
-		list(mode_name="RAPID", projectile_type=/obj/item/projectile/beam/pulse, fire_sound='sound/weapons/pulse.ogg', fire_delay=10, charge_cost=175, icon="destroy"),
+		list(mode_name="DESTROY", projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=null, charge_cost=null, icon="destroy"),
+		list(mode_name="RAPID", projectile_type=/obj/item/projectile/plasma/heavy, fire_sound='sound/weapons/pulse.ogg', fire_delay=10, charge_cost=175, icon="destroy"),
 	)
 
 
-/obj/item/weapon/gun/energy/pulse/cassad
+/obj/item/weapon/gun/energy/plasma/cassad
 	name = "FS PR \"Cassad\""
 	desc = "\"Frozen Star\" brand energy assault rifle, capable of prolonged combat. When surrender is not an option."
```

Comment:
"plasma assault rifle" maybe?

Reply:
It still technically consumes energy, but sure